[DRAFT] PX4 Mainatiner Role Documentation (#2313)
PX4/PX4-user_guide: en/contribute/maintainers.md

# Maintainer Role

To support PX4 Ecosystem, we have [maintainers](#role-description) for subsystems within PX4, as well as in other ecosystem components such as MAVSDK, QGroundControl 

PX4 maintainer role is organized by [Dronecode](https://www.dronecode.org/), and the detailed description can be found in [the website](TODO).

<!-- TODO: Add Dronecode PX4 Maintainers Role Description Link -->

> To find the most up-to-date maintainers structure, visit [PX4-Autopilot README](https://github.com/PX4/PX4-Autopilot#maintenance-team). <!-- NOTE: Need to insert updated link after https://github.com/PX4/PX4-Autopilot/pull/21030 gets merged -->

Author: Waiting for merge of https://github.com/PX4/PX4-Autopilot/pull/21030

# Maintainer Role

[PX4 maintainers](#role-description) have technical leadership and responsibility for specific areas of PX4, and for other ecosystem components such as MAVSDK, and QGroundControl.

> The role is organized by [Dronecode](https://www.dronecode.org/), and the detailed description can be found in the [role description](#role-description) section below.

Reviewer: I did a slight subedit and will merge.

  Attendees are encouraged to add to the agenda as a reply before the meeting starts.
* To nominate Issues and PRs for the call you can use the [maintainers-call](https://github.com/PX4/PX4-Autopilot/labels/maintainers-call) label to flag them

## Recruitment Process

If you would like to join the PX4 maintainers team:

1. Read the [role description](#dronecode-maintainer-role-description), and make sure you understand the responsibilities of the role.
2. Contact the relevant maintainer listed in the [PX4-Autopilot README](https://github.com/PX4/PX4-Autopilot#maintenance-team) on [Discord server](support.md#forums-and-chat), or any of the preferred method.
3. Express your interest in becoming a maintainer, and specify which area you would like to maintain.
4. Bring this up in the [weekly maintainers call](#maintainers-call).
   If there's a general consensus you will be approved as maintainer!

## Onboarding Process

Maintainers go through the following onboarding process:

1. **Discord** server admin will grant you the `dev team` role, which gives you:
   1. Permission to speak during maintainers call.
   2. Access to the `#maintainers` channel.

>    3. Access to the user group / dronecode member companies channels.

Reviewer: ```suggestion
```

### Responsibilities

1. Take charge of overseeing the development in their category.
2. Provide guidance/advice on community members in their category.
3. Help with PR review and Issues in their category.
4. Coordinate with the maintainer group.
5. Attend the weekly [maintainers meeting](#maintainers-call).
6. Help create and maintain a Roadmap for PX4.
7. Adhere to Dronecode's [Code of Conduct](https://github.com/Dronecode/foundation/blob/main/CODE-OF-CONDUCT.md).

### Qualifications

1. Experience in the category field with proven track record of contributions.
2. Good overview of the category in PX4 ecosystem.
3. Approval from your employer.

### Perks

1. **Official recognition** as the maintainer in Dronecode/PX4 website, documentation, community and social media.
2. **Github & Discord privileges** (described in the [onboarding process](#onboarding-process)).
3. Free ticket to the yearly **PX4 Dev Summit** and travel assistance.

### Tools we Provide to Assist You

Dronecode will provide the following tools to help you:

1. **Flight testing**: If you need a tester for your feature, we can find them and provide you with feedback/logs.

> 2. **Release schedule**: Clear timeline of upcoming releases, feature freezes to help you decide on deciding on development process.

Reviewer: ```suggestion
```

A detailed description can be found in the [role description](#role-description) section below.

To find the most up-to-date maintainers list, visit [PX4-Autopilot README](https://github.com/PX4/PX4-Autopilot#maintenance-team).

## Maintainers Call

The weekly maintainers call discusses overall PX4 structure, community and ecosystem.
You can find the relevant information below:

* Time: Tuesday 17h00 CET ([subscribe to calendar](https://www.dronecode.org/calendar/)).
* Place: [Dev-Call-Maintainers channel in Discord](https://discord.gg/RjxYXFSnj3).
* Meeting note is published week before in [Discuss Forum - PX4 Coordination](https://discuss.px4.io/c/weekly-dev-call/px4-dev-call/39).
  Attendees are encouraged to add to the agenda as a reply before the meeting starts.
* To nominate Issues and PRs for the call you can use the [maintainers-call](https://github.com/PX4/PX4-Autopilot/labels/maintainers-call) label to flag them

## Recruitment Process

If you would like to join the PX4 maintainers team:

1. Read the [role description](#dronecode-maintainer-role-description), and make sure you understand the responsibilities of the role.

> 2. To nominate yourself, reach out to one of the maintainers (See the complete list in the [PX4-Autopilot README](https://github.com/PX4/PX4-Autopilot#maintenance-team)), and seek their sponsorship.

Reviewer: ```suggestion
2. To nominate yourself, reach out to one of the maintainers (see the complete list in the [PX4-Autopilot README](https://github.com/PX4/PX4-Autopilot#maintenance-team)), and seek their sponsorship.
```

### Responsibilities

1. Take charge of overseeing the development in their category.
2. Provide guidance/advice on community members in their category.
3. Review relevant pull requests and issues from the community on GitHub.
4. Coordinate with the maintainer group.
5. Keep regular attendance at the [maintainers meeting](#maintainers-call).
6. Help create and maintain a roadmap for the project your represent.
7. Uphold the [Code of Conduct](https://github.com/Dronecode/foundation/blob/main/CODE-OF-CONDUCT.md) of our community.

### Qualifications

1. Proven track record of valuable contributions.
2. Domain expertise in the category field.
3. Good overview of the project you are applying to.
4. You need to manage approval from your employer when relevant.

### Perks

1. **Official recognition** as the maintainer in Dronecode/PX4 website, documentation, community and social media.
2. **Github & Discord privileges** (described in the [onboarding process](#onboarding-process)).
3. Priority placement to the yearly **PX4 Developer Summit** scholarship which helps you cover travel.

### Tools we Provide to Assist You

Dronecode will provide the following tools to help you:

> 1. **Flight testing**: If you need a tester for your feature, we can find them and provide you with feedback/logs.
> 3. **Community survey**: If you need any insight into the community's opinion, we will send out social media posts, mailing lists, announcements in Discord server to get that answer for you.
> 4. **Workflow automation**: We will provide workflow for PR/Issue review & tagging process to help you.

Reviewer: ```suggestion
1. **Community survey**: If you need any insight into the community's opinion, we will send out social media posts, mailing lists, announcements in Discord server to get that answer for you.
1. **Workflow automation**: We will provide workflow for PR/Issue review & tagging process to help you.
```

5. If approved, go through the [onboarding process](#onboarding-process)

## Onboarding process

After you become a maintainer, you will go through the following process:

1. **Discord** server admin will grant you the `dev team` role, which gives you:
   1. Permission to speak during maintainers call
   2. Access to the `#maintainers` channel
   3. Access to the user group / dronecode member companies channels
2. **Github** PX4 Workspace admin will grant you the [`Dev Team`](https://github.com/orgs/PX4/teams/dev-team) role, which gives you:
   1. Permission to merge the PR of any of PX4 workspace repositories after it's approved
   2. Permission to trigger Github actions when a new contributor opens a PR
   3. Permission to edit Issue/PR contents
3. **Dronecode member** (Junwoo/Ramon) will:
   1. Update the PX4 code repository [README](https://github.com/PX4/PX4-Autopilot/blob/main/README.md) to include your information
   2. Update the PX4 website [Maintainers section](https://px4.io/community/maintainers/) to include your information
   3. Add your information to the internal Dronecode database of maintainers to keep you in sync

>    4. Create a social media post about your new maintainers role after a short interview

Author: I have defined the clear steps of the recruitment & onboarding process here @mrpollo

## Dronecode Maintainer Role Description

### Summary

Dronecode Maintainer will lead/manage the development of a **specific category (referred as category below)** of the PX4 ecosystem. With this role, you will be at the epicenter of developing the world’s top most popular flight control software system!

### Responsibilities

1. Take charge of overseeing the development in their category
2. Provide guidance/advice on community members in their category
3. Help with PR review and Issues in their category
4. Coordinate with the maintainer group
5. Attend the weekly [Maintainers meeting](#maintainers-call)
6. Help create and maintain a Roadmap for PX4
7. Adhere to Dronecode's [Code of Conduct](https://github.com/Dronecode/foundation/blob/main/CODE-OF-CONDUCT.md)

### Qualifications

1. Experience in the category field with proven track record of contributions
2. Good overview of the category in PX4 ecosystem

> 3. Approval from your employer

Author: As @Jaeyoung-Lim, since Maintainers role isn't paid, I don't think this is necessary. What do you think? (Everyone)

Reviewer: I don't think it is necessary because this should not be taking up "work time".

### Perks

1. **Official recognition** as the maintainer in Dronecode/PX4 website, documentation, community and social media
2. **Github & Discord privileges** (described in the [onboarding process](#onboarding-process))

> 3. Free ticket to the yearly **PX4 Dev Summit** and travel assistance

Author: Shall we clarify this a bit more, or shall we keep it like this (up to interpretation / person by person case)? @mrpollo

Reviewer: Perhaps this - it implies the "not guaranteed" nature.

```suggestion
3. Free ticket to the yearly **PX4 Dev Summit** and preferential travel assistance
```

### Tools we provide to assist you

To help you as a maintainer, Dronecode will provide:

1. **Flight testing**: If you need a tester for your feature, we can find them and provide you with feedback/logs

> 2. **Release schedule**: Clear timeline of upcoming releases, feature freezes to help you decide on deciding on development process

Reviewer: Really, I've never had this?

Reviewer 2: All of this was in a google doc as a draft, and wasn't really discussed, I asked @junwoo091400 to move things over to a PR, and didn't review beforehand, I'll make a few suggestions

Reviewer: Sound good!

Author: This is now in progress here: https://github.com/PX4/PX4-user_guide/pull/2331

3. **Community survey**: If you need any insight into the community's opinion, we will send out social media posts, mailing lists, announcements in Discord server to get that answer for you
4. **Workflow automation**: We will provide workflow for PR/Issue review & tagging process to help you

And as always, don't hesitate to reach out if you need help with anything. We are here for you!

### Point of Contact

Regarding questions about the Maintainers role, please contact **Ramon** in discord server or by [email](rroche@linuxfoundation.org).

> ### Revision Notes
> 
> | Date | Version | Description | Editor |
> |---|---|---|---|
> | Feb 28th 2023 | 1.0.0 | First revision updated in Documentation repository | Junwoo Hwang |

Reviewer: We don't do this for anything else.

```suggestion
```

Author: Oops sorry I missed this!

# Maintainer Role

[PX4 maintainers](#role-description) have technical leadership and responsibility for specific areas of PX4, and for other ecosystem components such as MAVSDK, and QGroundControl.

The role is organized by [Dronecode](https://www.dronecode.org/), and the detailed description can be found in the [role description](#role-description) section below.

To find the most up-to-date maintainers list, visit [PX4-Autopilot README](https://github.com/PX4/PX4-Autopilot#maintenance-team).

## Maintainers Call

The weekly maintainers call discusses overall PX4 structure, community and ecosystem.
You can find the relevant information below:

> * Time: Tuesday 17h00 CET ([subscribe to calendar](https://www.dronecode.org/calendar/)).

Reviewer: Is three a maintainer call?

* Place: [Dev-Call-Maintainers channel in Discord](https://discord.gg/RjxYXFSnj3).
* Meeting note is published week before in [Discuss Forum - PX4 Coordination](https://discuss.px4.io/c/weekly-dev-call/px4-dev-call/39).
  Attendees are encouraged to add to the agenda as a reply before the meeting starts.
* To nominate Issues and PRs for the call you can use the [maintainers-call](https://github.com/PX4/PX4-Autopilot/labels/maintainers-call) label to flag them

## Recruitment Process

If you would like to join the PX4 maintainers team:

1. Read the [role description](#dronecode-maintainer-role-description), and make sure you understand the responsibilities of the role.
2. Contact the relevant maintainer listed in the [PX4-Autopilot README](https://github.com/PX4/PX4-Autopilot#maintenance-team) on [Discord server](support.md#forums-and-chat), or any of the preferred method.
3. Express your interest in becoming a maintainer, and specify which area you would like to maintain.

> 4. Bring this up in the [weekly maintainers call](#maintainers-call).

Reviewer: ```suggestion
4. Your sponsoring maintainer needs to bring this up for discussion in one of the [weekly maintainers calls](#maintainers-call).
```

Maintainers go through the following onboarding process:

1. **Discord** server admin will grant you the `dev team` role, which gives you:
   1. Permission to speak during maintainers call.
   2. Access to the `#maintainers` channel.
   3. Access to the user group / dronecode member companies channels.
2. **Github** PX4 Workspace admin will grant you the [`Dev Team`](https://github.com/orgs/PX4/teams/dev-team) role, which gives you:
   1. Permission to merge the PR of any of PX4 workspace repositories after it's approved
   2. Permission to trigger Github actions when a new contributor opens a PR.
   3. Permission to edit Issue/PR contents.
3. **Dronecode member** (Junwoo/Ramon) will:
   1. Update the PX4 code repository [README](https://github.com/PX4/PX4-Autopilot/blob/main/README.md) to include your information.
   2. Update the PX4 website [Maintainers section](https://px4.io/community/maintainers/) to include your information.
   3. Add your information to the internal Dronecode database of maintainers to keep you in sync.
   4. Create a social media post about your new maintainers role after a short interview.

## Dronecode Maintainer Role Description

### Summary

> Dronecode Maintainer will lead/manage the development of a **specific category (referred as category below)** of the PX4 ecosystem. With this role, you will be at the epicenter of developing the world’s top most popular flight control software system!

Reviewer: Maintainer is a description, not a proper noun here.

```suggestion
Maintainers lead/manage the development of a **specific category (referred to as category below)** of any Open Source Projects hosted within the Dronecode Foundation, such as the PX4 Autopilot. 
```

### Perks

1. **Official recognition** as the maintainer in Dronecode/PX4 website, documentation, community and social media.
2. **Github & Discord privileges** (described in the [onboarding process](#onboarding-process)).
3. Free ticket to the yearly **PX4 Dev Summit** and travel assistance.

### Tools we Provide to Assist You

Dronecode will provide the following tools to help you:

1. **Flight testing**: If you need a tester for your feature, we can find them and provide you with feedback/logs.
2. **Release schedule**: Clear timeline of upcoming releases, feature freezes to help you decide on deciding on development process.
3. **Community survey**: If you need any insight into the community's opinion, we will send out social media posts, mailing lists, announcements in Discord server to get that answer for you.
4. **Workflow automation**: We will provide workflow for PR/Issue review & tagging process to help you.

And as always, don't hesitate to reach out if you need help with anything. We are here for you!

### Point of Contact

Regarding questions about the maintainer role, please contact **Ramon** in discord server or by [email](rroche@linuxfoundation.org).

> ### Revision Notes
> 
> | Date | Version | Description | Editor |
> |---|---|---|---|
> | Feb 28th 2023 | 1.0.0 | First revision updated in Documentation repository | Junwoo Hwang |

Reviewer: Closing is not resolving.

```suggestion
```
